pandas/core/reshape/concat.py

```diff
@@ -705,7 +705,7 @@ def _make_concat_multiindex(indexes, keys, levels=None, names=None) -> MultiInde
             names = [None]
 
         if levels is None:
-            levels = [ensure_index(keys)]
+            levels = [ensure_index(keys).unique()]
```

Comment:
hmm shouldn't this be the case for a specified levels as well?

Reply:
Can we check whether the level is unique before? If not, raise ValueError. The doc says it should be unique.

Reply:
I find actually we actually do not have the check for depulicated levels in concat function. Something like the following will not raise. Since this problem is an isolated one, thus I will make another PR to escape from confusion.

```python
df1 = pd.DataFrame({"A": [1]}, index=["x"])
df2 = pd.DataFrame({"A": [1]}, index=["y"])
pd.concat([df1, df2], levels=[["x", "y", "y"]]) # should raise
```

Reply:
@GYHHAHA - Agreed what you are pointing out is a separate issue, but here is an example that @jreback was referring to.

```
df1 = pd.DataFrame({"x": [1, 2], "y": [3, 4], "z": [5, 6]}).set_index(["x", "y"])
df2 = pd.DataFrame({"x": [7, 8], "y": [9, 10], "z": [11, 12]}).set_index(["x", "y"])
result = pd.concat([df1, df2, df1], keys=["x", "y", "x"], levels=[["x", "y", "x"]])  # should raise
print(result.loc['x', '1', '3'])
```

This also raises the same error that is being addressed here.

Reply:
why you loc with string '1' and '3' instead of numeric value?

Reply:
Thanks - that was a typo; these should be integers.

Reply:
Actually if I replace them with integers, it doesn't raise. Could you check? Thanks. @rhshadrach 

```python
>>> df1 = pd.DataFrame({"x": [1, 2], "y": [3, 4], "z": [5, 6]}).set_index(["x", "y"])
>>> df2 = pd.DataFrame({"x": [7, 8], "y": [9, 10], "z": [11, 12]}).set_index(["x", "y"])
>>> result = pd.concat([df1, df2, df1], keys=["x", "y", "x"], levels=[["x", "y", "z"]])
[['x', 'y', 'z']]
>>> result.loc['x', 1, 3]
<stdin>:1: PerformanceWarning: indexing past lexsort depth may impact performance.
          z
   x y
x 1 3  5
      3  5
```

Reply:
On the `result = ...` line, levels should be `[["x", "y", "x"]]`; not "z".

Reply:
Yeah, now I get the error. I will look into this.

Reply:
Sounds good! I believe you just need to apply your change to the `else` clause highlighted here (but I could be wrong).

Reply:
But I believe this is caused by duplicated levels input, if levels is [["x", "y"]], then it works fine. Maybe more suitable to add this to another PR related to unique levels keyword. @rhshadrach